migration/v0.34/apply/11-prometheus-operator-crds.sh

```diff
@@ -9,17 +9,21 @@ run() {
-      kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-servicemonitors.yaml
-      kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-thanosrulers.yaml
-      kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-scrapeconfigs.yaml
-      kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-prometheusagents.yaml
+      if [[ ! "$(helm_chart_version "$CLUSTER" monitoring kube-prometheus-stack)" = "49.2.0" ]]; then
+        log_info "  - applying the prometheus-operator CRDs on $CLUSTER"
+        kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-alertmanagerconfigs.yaml
+        kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-alertmanagers.yaml
+        kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-podmonitors.yaml
+        kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-probes.yaml
+        kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-prometheuses.yaml
+        kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-prometheusrules.yaml
+        kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-servicemonitors.yaml
+        kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-thanosrulers.yaml
+        kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-scrapeconfigs.yaml
+        kubectl_do $CLUSTER apply --server-side --force-conflicts -f "${ROOT}"/helmfile/upstream/prometheus-community/kube-prometheus-stack/charts/crds/crds/crd-prometheusagents.yaml
+      else
+        log_info "  - Skipping applying the prometheus-operator CRDs on $CLUSTER"
+      fi
     done
     ;;
   rollback)
```

Comment:
You should do the same check for the rollback so it won't accidentally revert if kube-prometheus-stack already was updated. The probability of hitting that scenario is quite low, but not zero.